src/xdp-utils.c

```diff
@@ -144,11 +145,72 @@ xdp_app_info_new (XdpAppInfoKind kind)
   return app_info;
 }
 
+void
+decode_hex_escapes (char *str[])
+{
+  char hex[] = "0xhh";
+
+  for (int i=0; (*str)[i]!='\0' && (*str)[i+1]!='\0'; ++i)
+  {
+    if ((*str)[i] == '\\' && (*str)[i+1] == 'x')
+      {
+        hex[2] = (*str)[i+2];
+        hex[3] = (*str)[i+3];
+        char unescaped = (char) strtol (hex, NULL, 16);
+        (*str)[i+2] = '\\';
+        (*str)[i+3] = unescaped;
+        i += 3;
+      }
+  }
+  char **splits = g_strsplit(*str, "\\x\\", -1);
+  g_free (*str);
+  *str = g_strjoinv ("", splits);
+  g_strfreev (splits);
```

Comment:
After going to heroic lengths to do the decoding in-place with no extra allocations, this does a lot of malloc/free cycles at the end...

I think a better approach to this would be to take a `const char *` argument and iterate through it, appending characters to a `GString` buffer, returning a newly allocated string with `g_string_free (gstr, FALSE)`.